be/src/storage/rowset/segment_iterator.cpp

```diff
@@ -392,9 +392,13 @@ Status SegmentIterator::_init() {
     RETURN_IF_ERROR(_get_row_ranges_by_keys());
     RETURN_IF_ERROR(_get_row_ranges_by_rowid_range());
     RETURN_IF_ERROR(_apply_del_vector());
-    RETURN_IF_ERROR(_apply_bitmap_index());
-    RETURN_IF_ERROR(_get_row_ranges_by_zone_map());
-    RETURN_IF_ERROR(_get_row_ranges_by_bloom_filter());
+    // skip some index filters if the scan range is already empty or single row.
+    // single row case happens when query is a pointer query and we already use pk index to get a single row range
+    if (!_scan_range.is_single_row_or_empty()) {
```

Comment:
If the query is a point query, the `bitmap index` can helps for not primary key table.

Reply:
in that case, `!_scan_range.is_single_row_or_empty()` will be true

Reply:
not primary key table, `!_scan_range.is_single_row_or_empty()` will be true

Reply:
removed